packages/engine/paima-funnel/src/index.ts

```diff
@@ -84,27 +88,53 @@ export class FunnelFactory implements IFunnelFactory {
     // start with a base funnel
     // and wrap it with dynamic decorators as needed
     let chainFunnel: ChainFunnel = await BlockFunnel.recoverState(this.sharedData, dbTx);
-    for (const [chainName, config] of await GlobalConfig.otherEvmConfig()) {
-      chainFunnel = await wrapToParallelEvmFunnel(
-        chainFunnel,
-        this.sharedData,
-        dbTx,
-        ENV.START_BLOCKHEIGHT,
-        chainName,
-        config
-      );
-    }
-    chainFunnel = await wrapToCarpFunnel(chainFunnel, this.sharedData, dbTx, ENV.START_BLOCKHEIGHT);
-    for (const [chainName, config] of await GlobalConfig.minaConfig()) {
-      chainFunnel = await wrapToMinaFunnel(
-        chainFunnel,
-        this.sharedData,
-        dbTx,
-        ENV.START_BLOCKHEIGHT,
-        chainName,
-        config
-      );
+
+    const configs = await GlobalConfig.getInstance();
+
+    for (const chainName of Object.keys(configs)) {
+      const config = configs[chainName];
+      const type = config.type;
+
+      switch (type) {
+        case ConfigNetworkType.EVM_OTHER:
+          chainFunnel = await wrapToParallelEvmFunnel(
+            chainFunnel,
+            this.sharedData,
+            dbTx,
+            ENV.START_BLOCKHEIGHT,
+            chainName,
+            config as OtherEvmConfig
+          );
+          break;
+        case ConfigNetworkType.MINA:
+          chainFunnel = await wrapToMinaFunnel(
+            chainFunnel,
+            this.sharedData,
+            dbTx,
+            ENV.START_BLOCKHEIGHT,
+            chainName,
+            config as MinaConfig
+          );
+          break;
+        case ConfigNetworkType.CARDANO:
+          chainFunnel = await wrapToCarpFunnel(
+            chainFunnel,
+            this.sharedData,
+            dbTx,
+            ENV.START_BLOCKHEIGHT,
+            chainName,
+            config as CardanoConfig
+          );
+          break;
+        case ConfigNetworkType.EVM:
+          // the base funnel has to go first (even if it's not the first in the
+          // order) and it's already initialized
```

Comment:
Is this still true with the Avail branch being merged?

Reply:
I think so. I wrote base funnel instead of block funnel so that it's general. If the avail block funnel is used then it also goes first.